### PR TITLE
Adding the Encoding Service to shared code

### PR DESCRIPTION
**Pull request by guypritchard**
We use this in a few places, might as well have one implementation now we have a library for such things.

### CODE REVIEW

File: src/RequestProcessor/Encoder/EncodingService.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            };
+
+            Span<byte> buffer = new(new byte[content.Length]);
+            var isBase64 = Convert.TryFromBase64String(content, buffer, out int bytesWritten);
+            if (isBase64)
+            {
+                var s = Encoding.UTF8.GetString(buffer[..bytesWritten]);
```
Comment (joonasroosalung) [inline src/RequestProcessor/Encoder/EncodingService.cs:18]: Why not just return on this line?

File: src/RequestProcessor/Encoder/EncodingService.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        public string EncodeBase64(string content)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            };
+
+            byte[] bytes = Encoding.UTF8.GetBytes(content);
+            var base64 = Convert.ToBase64String(bytes);
```
Comment (joonasroosalung) [inline src/RequestProcessor/Encoder/EncodingService.cs:35]: Also could return right away on this line, perhaps?

**Review by joonasroosalung — APPROVED** [review]
Two small comments

File: src/RequestProcessor/Encoder/EncodingService.cs (hunk 1 above)
Comment (guypritchard) [reply src/RequestProcessor/Encoder/EncodingService.cs:18]: Will do this in a subsequent change - I suspect it looks like this so someone can see the value more easily in a debugger.

**Pull request merged**